lib/mgmt_be_client.c

```diff
@@ -483,58 +424,78 @@ static int mgmt_be_txn_cfg_prepare(struct mgmt_be_txn_ctx *txn, const char **con
 	bool error;
 	char err_buf[BUFSIZ];
 	size_t num_processed;
+	bool disconnect = false;
 	int err;
 
-	assert(txn && txn->client);
-	client_ctx = txn->client;
-
 	num_processed = 0;
 	error = false;
 	nb_ctx.client = NB_CLIENT_CLI;
 	nb_ctx.user = (void *)client_ctx->user_data;
 	err_buf[0] = 0;
 
+	debug_be_client("Creating new txn-id %Lu", txn_id);
+
 	/*
-	 * this is always true, right?!
+	 * First validate there's no other config txn right now. Eventually we
+	 * may want to support multiple CFG_REQ prioer to a CFG_APPLY.
 	 */
-	assert(!txn->nb_txn);
+	if (client_ctx->config_txn) {
+		log_err_be_client("Cannot prepare cfg for txn-id: %Lu, another txn-id: %Lu was in progress",
+				  txn_id, client_ctx->config_txn->txn_id);
+		return true;
+	}
 
-	if (!txn->nb_txn) {
-		/*
-		 * XXX, is this just an old crusty comment?
-		 *
-		 * This happens when the current backend client is only
-		 * interested in consuming the config items but is not
-		 * interested in validating it.
-		 */
-		gettimeofday(&edit_nb_cfg_start, NULL);
-
-		error = be_client_change_edit_config(client_ctx->candidate_config, config, err_buf,
-						     sizeof(err_buf));
-		if (error) {
-			log_err_be_client("Failed to update configs for txn-id: %" PRIu64
-					  " to candidate, err: '%s'",
-					  txn->txn_id, err_buf);
-			return -1;
-		}
+	/* Create a txn to track the config changes for later apply */
+	txn = mgmt_be_txn_create(client_ctx, txn_id);
+	if (!txn) {
```

Comment:
So in this case txn is NULL and we goto done; and in done we call mgmt_be_txn_cfg_abort whidh asserts that txn is non NULL.  Won't this crash?